extensions/amp-fx-flying-carpet/0.1/amp-fx-flying-carpet.js

```diff
@@ -110,15 +110,15 @@ class AmpFlyingCarpet extends AMP.BaseElement {
     const docHeight = viewport.getScrollHeight();
     // Hmm, can the page height change and affect us?
     user().assert(
-      layoutBox.top >= viewportHeight,
+      layoutBox.top >= viewportHeight * 0.8,
```

Comment:
Worse case is actually when it appears on the first viewport and does not touch the bottom of the viewport (so the whole carpet is visible in first viewport). Then, you'll never see the bottom of the content.

No documentation was written, just an issue you brought up in hangouts.